client/client.go

```diff
@@ -0,0 +1,227 @@
+package client
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/prometheus-community/pushprox/util"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	scrapeErrorCounter = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "pushprox_client_scrape_errors_total",
+			Help: "Number of scrape errors",
+		},
+	)
+	pushErrorCounter = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "pushprox_client_push_errors_total",
+			Help: "Number of push errors",
+		},
+	)
+	pollErrorCounter = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "pushprox_client_poll_errors_total",
+			Help: "Number of poll errors",
+		},
+	)
+)
+
+func init() {
+	prometheus.MustRegister(pushErrorCounter, pollErrorCounter, scrapeErrorCounter)
+}
+
+func DefaultBackoff() backoff.BackOff {
+	b := backoff.NewExponentialBackOff()
+	b.InitialInterval = 1 * time.Second
+	b.Multiplier = 1.5
+	b.MaxInterval = 5 * time.Second
+	b.MaxElapsedTime = time.Duration(0)
+	return b
+}
+
+// Coordinator for scrape requests and responses
+type Coordinator struct {
```

Comment:
Coordinator -> ScrapingCoordinator
then the comment will not be needed and it will be indicative when looking at an instance of it